x-pack/plugins/security/server/lib/auth_redirect.js

```diff
@@ -33,18 +33,27 @@ export function authenticateFactory(server) {
 
     if (authenticationResult.succeeded()) {
       return h.authenticated({ credentials: authenticationResult.user });
-    } else if (authenticationResult.redirected()) {
+    }
+
+    if (authenticationResult.redirected()) {
       // Some authentication mechanisms may require user to be redirected to another location to
       // initiate or complete authentication flow. It can be Kibana own login page for basic
       // authentication (username and password) or arbitrary external page managed by 3rd party
       // Identity Provider for SSO authentication mechanisms. Authentication provider is the one who
       // decides what location user should be redirected to.
       return h.redirect(authenticationResult.redirectURL).takeover();
-    } else if (authenticationResult.failed()) {
+    }
+
+    if (authenticationResult.failed()) {
       server.log(['info', 'authentication'], `Authentication attempt failed: ${authenticationResult.error.message}`);
-      return wrapError(authenticationResult.error);
-    } else {
-      return Boom.unauthorized();
+      const error = wrapError(authenticationResult.error);
```

Comment:
__note:__ we could potentially move this `if` and optional `challenges` to `wrapError`, but I'm leaning towards targeted usage at this stage.

Reply:
I like the explicitness of how it's done currently.